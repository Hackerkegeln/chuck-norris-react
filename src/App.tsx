import React from 'react';
import logo from './logo.jpg';
import {Result} from './Result';

const App: React.FC = () => {
  const refreshJoke = () => {
    alert('Works!');
  };

  return (
    <div className="App, container-fluid">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo"/>
      </header>
      <main>
        <Result jokeText={"Hello, World!"}/>
      </main>
      <footer>
        <button className="btn btn-primary" onClick={refreshJoke}>Refresh</button>
      </footer>
    </div>
  );
};

export default App;
