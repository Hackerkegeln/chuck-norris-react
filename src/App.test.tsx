import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

test('renders the image', () => {
  const { getByRole } = render(<App />);
  const banner = getByRole('img');
  expect(banner).toHaveAttribute('src', 'logo.jpg');
  expect(banner).toHaveAttribute('alt', 'logo');
});

test('renders the page by snapshot', () => {
  const {container} = render(<App />);
  expect(container).toMatchSnapshot();
});
