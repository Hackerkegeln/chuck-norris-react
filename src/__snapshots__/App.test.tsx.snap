// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the page by snapshot 1`] = `
<div>
  <div
    class="App, container-fluid"
  >
    <header
      class="App-header"
    >
      <img
        alt="logo"
        class="App-logo"
        src="logo.jpg"
      />
    </header>
    <main>
      <div
        class="spinner-border"
        role="status"
      >
        <span
          class="sr-only"
        >
          Loading...
        </span>
      </div>
    </main>
    <footer>
      <button
        class="btn btn-primary"
      >
        Refresh
      </button>
    </footer>
  </div>
</div>
`;
